Caboodle/Shared/Exceptions.shared.cs

```diff
@@ -2,11 +2,19 @@
 
 namespace Microsoft.Caboodle
 {
-    public class NotImplentedInReferenceAssembly : NotImplementedException
+    public class NotImplentedInReferenceAssemblyException : NotImplementedException
```

Comment:
Will it be possible to move the rename into a separate PR so as not to mix things? Not essential, but something nice.

Reply:
yup